components/script/dom/htmlelement.rs

```diff
@@ -839,10 +904,97 @@ impl VirtualMethods for HTMLElement {
                     DOMString::from(&**attr.value()),
                 );
             },
+            (&local_name!("form"), mutation) => {
+                if self.is_form_associated_custom_element() {
+                    self.form_attribute_mutated(mutation);
+                }
+            },
+            (&local_name!("disabled"), AttributeMutation::Set(_)) => {
+                // Adding a "disabled" attribute disables an enabled form element.
+                if self.is_form_associated_custom_element() {
+                    let el = self.upcast::<Element>();
```

Comment:
I would move this outside the match to avoid repetition, but please spell out "element":
```suggestion
                    let element = self.upcast::<Element>();
```